src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1483,22 +1483,26 @@ reg_t MPS::apply_measure_internal(const reg_t &qubits,
 				  RngEngine &rng) {
   // When all qubits are measured, then for every qubit measured, it is sufficient to 
   // propagate to the nearest neighbors because the neighbors will be measured next
-  bool measure_all = 0;
-  if (qubits.size() == num_qubits_)
-    measure_all = true;
   reg_t qubits_to_update;
-  reg_t outcome_vector(qubits.size());
-  for (uint_t i=0; i<qubits.size(); i++) {
+  uint_t size = qubits.size();
+  reg_t outcome_vector(size);
+  reg_t sorted_qubits = qubits;
+  std::sort(sorted_qubits.begin(), sorted_qubits.end());
+
+  bool measure_right_neighbor = false;
+  for (uint_t i=0; i<size; i++) {
+    measure_right_neighbor = (i<size-1 && sorted_qubits[i+1] == i+1);
+
```

Comment:
Please add inline comments with explanations and examples

Reply:
Added documentation. Let me know if it is clear now.